sdk/zklogin/src/checks.ts

```diff
@@ -0,0 +1,282 @@
+		if (ext_claim_end_event.id !== 'onSeparator' && ext_claim_end_event.id !== 'onObjectEnd') {
+			throw new Error(`Unexpected error: Unexpected ext_claim_end_event. 
+                             Event: ${JSON.stringify(ext_claim_end_event)}`);
+		}
+
+		if (value_event.arg === undefined) {
+			throw new Error(`Unexpected error: Undefined value_event.arg?. 
+                             Event: ${JSON.stringify(value_event)}`);
+		}
+		this.processed[name] = {
+			name: name,
+			value: value_event.arg,
+			ext_claim: this.decoded_payload.slice(name_event.offset, ext_claim_end_event.offset + 1),
+			offsets: {
+				start: name_event.offset,
+				colon: colon_event.offset - name_event.offset,
+				value: value_event.offset - name_event.offset,
+				value_length: value_event.length,
+				length: ext_claim_end_event.offset - name_event.offset + 1,
+			},
+		};
+
+		// The number of whitespaces is equal to the length of extended claim
+		//  minus the length of the value minus the length of the name minus 2.
+		//  (2 for the colon and either comma or a close brace)
+		const num_whitespaces =
+			this.processed[name].offsets.length - value_event.length - name_event.length - 2;
+		if (num_whitespaces > 0) {
+            // TODO: This is an interesting event to note.
+            //       Note sure if the console.info is the right way to log it.
+			console.info(`[Rare event] Non-zero whitespace detected: 
+                          Claim ${name} has ${num_whitespaces} whitespaces`);
+		}
+
+		return this.processed[name];
+	}
+
+	/**
+	 * Returns the claim value exactly as it appears in the JWT.
+	 * So, if it has escapes, no unescaping is done.
+	 * Assumes that the claim value is a string.
+	 * (reasonable as aud and common key claims like sub, email and username are JSON strings)
+	 *
+	 * @param name claim name
+	 * @returns claim value as it appears in the JWT without the quotes. The quotes are omitted to faciliate address derivation.
+	 *
+	 * NOTE: This function is only used to obtain claim values for address generation.
+	 * Do not use it elsewhere unless you know what you're doing.
+	 */
+	getRawClaimValue(name: string): string {
+		if (!Object.prototype.hasOwnProperty.call(this.processed, name)) {
+			throw new Error('Claim ' + name + ' not processed');
+		}
+		const details = this.processed[name];
+
+		const value_index = details.offsets.value + details.offsets.start;
+		const value_length = details.offsets.value_length;
+		if (this.decoded_payload[value_index] !== '"') {
+			throw new Error(
+				`Claim ${name} does not have a string value. Details: ${JSON.stringify(details)}`,
+			);
+		}
+		if (this.decoded_payload[value_index + value_length - 1] !== '"') {
+			throw new Error(
+				`Claim ${name} does not have a string value. Details: ${JSON.stringify(details)}`,
+			);
+		}
+
+		const raw_value = this.decoded_payload.slice(value_index + 1, value_index + value_length - 1); // omit the quotes
+		if (raw_value !== details.value) {
```

Comment:
Is this an error? If it is lets through if not let's remove it, no need to add this console log